bundles/org.openhab.core.binding.xml/src/main/java/org/eclipse/smarthome/core/binding/xml/internal/XmlBindingInfoProvider.java

```diff
@@ -86,24 +92,15 @@ public void deactivate(ComponentContext componentContext) {
     }
 
     @Override
-    public synchronized BindingInfo getBindingInfo(String id, Locale locale) {
-        return get(id, locale);
+    public synchronized @Nullable BindingInfo getBindingInfo(@Nullable String id, @Nullable Locale locale) {
```

Comment:
The JavaDoc in the `BindingInfoProvider` interface says that it can be null or empty but I think that is a mistake.

https://github.com/openhab/openhab-core/blob/d39b6bd35ba231a0633bdf6f5ec6a0922f63db47/bundles/org.openhab.core/src/main/java/org/eclipse/smarthome/core/binding/BindingInfoProvider.java#L33

Reply:
I think this is in order to be "forgiving" to callers - if null is passed in, it won't break, but simply also return null.
But I think we can certainly change it here to expect a non-null id to be passed.